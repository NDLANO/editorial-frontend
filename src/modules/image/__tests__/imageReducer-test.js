/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import reducer, { actions } from '../image';

test('reducers/image initalState', () => {
  const nextState = reducer(undefined, { type: 'Noop' });

  expect(nextState).toMatchSnapshot();
});

test('reducers/image set image', () => {
  const nextState = reducer(undefined, {
    type: actions.setImage,
    payload: { id: 1, title: 'Unit test' },
  });

  expect(nextState).toMatchSnapshot();
});

test('reducers/image set multiple image', () => {
  const state = reducer(undefined, {
    type: actions.setImage,
    payload: { id: 1, title: 'Unit test 1' },
  });
  const nextState = reducer(state, {
    type: actions.setImage,
    payload: { id: 2, title: 'Unit test 2' },
  });

  expect(nextState).toMatchSnapshot();
});

test('reducers/image overwrite image with same id', () => {
  const nextState = reducer(
    {
      all: {
        1: { id: 1, title: 'Unit test 1' },
      },
    },
    { type: actions.setImage, payload: { id: 1, title: 'Unit test 2' } },
  );

  expect(nextState).toMatchSnapshot();
});

test('reducers/image sets isSaving to true on update', () => {
  const nextState = reducer(undefined, {
    type: actions.updateImage,
    payload: undefined,
  });

  expect(nextState).toMatchSnapshot();
});

test('reducers/image sets isSaving to false on update error', () => {
  const nextState = reducer(
    { isSaving: true },
    { type: actions.updateImageError, payload: undefined },
  );

  expect(nextState).toMatchSnapshot();
});

test('reducers/image sets isSaving to false on update success', () => {
  const nextState = reducer(
    { isSaving: true },
    { type: actions.updateImageuccess, payload: undefined },
  );

  expect(nextState).toMatchSnapshot();
});
