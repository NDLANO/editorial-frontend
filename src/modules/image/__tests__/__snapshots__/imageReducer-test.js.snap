// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/image initalState 1`] = `
Object {
  "all": Object {},
  "isSaving": false,
}
`;

exports[`reducers/image overwrite image with same id 1`] = `
Object {
  "all": Object {
    "1": Object {
      "id": 1,
      "title": "Unit test 2",
    },
  },
}
`;

exports[`reducers/image set image 1`] = `
Object {
  "all": Object {
    "1": Object {
      "id": 1,
      "title": "Unit test",
    },
  },
  "isSaving": false,
}
`;

exports[`reducers/image set multiple image 1`] = `
Object {
  "all": Object {
    "1": Object {
      "id": 1,
      "title": "Unit test 1",
    },
    "2": Object {
      "id": 2,
      "title": "Unit test 2",
    },
  },
  "isSaving": false,
}
`;

exports[`reducers/image sets isSaving to false on update error 1`] = `
Object {
  "isSaving": false,
}
`;

exports[`reducers/image sets isSaving to false on update success 1`] = `
Object {
  "isSaving": true,
}
`;

exports[`reducers/image sets isSaving to true on update 1`] = `
Object {
  "all": Object {},
  "isSaving": true,
}
`;
